patches/server/0977-Add-event-for-player-editing-sign.patch

```diff
@@ -0,0 +1,53 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: by77er <by77er@gmail.com>
+Date: Sat, 10 Jun 2023 19:09:03 -0400
+Subject: [PATCH] Add event for player editing sign
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/SignBlock.java b/src/main/java/net/minecraft/world/level/block/SignBlock.java
+index 91bb294be2cd8ab3467a62006a5a2751e0bec4ba..0f02bd1acaf12be255958ddf27d8c7993e15e873 100644
+--- a/src/main/java/net/minecraft/world/level/block/SignBlock.java
++++ b/src/main/java/net/minecraft/world/level/block/SignBlock.java
+@@ -76,9 +76,9 @@ public abstract class SignBlock extends BaseEntityBlock implements SimpleWaterlo
+     public InteractionResult use(BlockState state, Level world, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit) {
+         ItemStack itemStack = player.getItemInHand(hand);
+         Item item = itemStack.getItem();
+-        Item signBlockEntity = itemStack.getItem();
++        Item potentialApplicator = itemStack.getItem(); // Paper: renamed signBlockEntity -> potentialApplicator due to decompile error
+         SignApplicator var10000;
+-        if (signBlockEntity instanceof SignApplicator signApplicator) {
++        if (potentialApplicator instanceof SignApplicator signApplicator) { // Paper: renamed signBlockEntity -> potentialApplicator due to decompile error
+             var10000 = signApplicator;
+         } else {
+             var10000 = null;
+@@ -86,8 +86,8 @@ public abstract class SignBlock extends BaseEntityBlock implements SimpleWaterlo
+ 
+         SignApplicator signApplicator2 = var10000;
```

Comment:
This whole block is super confusing, idk what happened to the decompiler here. Is it worth just deleting it and rewriting it to make actual sense? Looking for input here, not saying it should be done.

Reply:
Yea I honestly wouldn't mind just yanking it.
I mean, its gonna get diffed on every update that touches it anyway with the amount of decompiler fixes it requires.

Reply:
Both 
```patch
-        Item signBlockEntity = itemStack.getItem();
-        SignApplicator var10000;
-        if (signBlockEntity instanceof SignApplicator signApplicator) {
-            var10000 = signApplicator;
-        } else {
-            var10000 = null;
-        }
-
-        SignApplicator signApplicator2 = var10000;
+        SignApplicator signApplicator2 = item instanceof SignApplicator signApplicator ? signApplicator : null; // Paper - decompile fixes
```

and 
```patch
-        BlockEntity bl2 = world.getBlockEntity(pos);
-        if (bl2 instanceof SignBlockEntity signBlockEntity) {
+        if (world.getBlockEntity(pos) instanceof SignBlockEntity signBlockEntity) { // Paper - decompile fixes
```
would probably do it.

Reply:
Added a commit that flattens the logic out in a more readable way. LMK if this is what you meant!

Reply:
Might be a bit too much imo. The two diffs I posted above yank a bunch but leave the core logic that follows in tact.
Tho I'd like machines input on this too in regards to what he wanted :+1: 

Thanks for the work already tho!

Reply:
Of course. Whatever works best!